lib/dependencies/AMDRequireDependenciesBlockParserPlugin.js

```diff
@@ -2,172 +2,175 @@
-					dep = null;
-					return true;
-				}
-				dep.functionBindThis = processFunctionArgument(this, expr.arguments[1]);
-				if(expr.arguments.length === 3) {
-					dep.errorCallbackBindThis = processFunctionArgument(this, expr.arguments[2]);
+			if(expr.arguments.length === 2 || expr.arguments.length === 3) {
+				try {
+					parser.inScope([], () => {
+						result = parser.applyPluginsBailResult("call require:amd:array", expr, param);
+					});
+					if(!result) {
+						dep = new UnsupportedDependency("unsupported", expr.range);
+						old.addDependency(dep);
+						if(parser.state.module)
+							parser.state.module.errors.push(new UnsupportedFeatureWarning(parser.state.module, "Cannot statically analyse 'require(..., ...)' in line " + expr.loc.start.line));
+						dep = null;
+						return true;
+					}
+					dep.functionBindThis = this.processFunctionArgument(parser, expr.arguments[1]);
+					if(expr.arguments.length === 3) {
+						dep.errorCallbackBindThis = this.processFunctionArgument(parser, expr.arguments[2]);
+					}
+				} finally {
+					parser.state.current = old;
+					if(dep)
+						parser.state.current.addBlock(dep);
 				}
-			} finally {
-				this.state.current = old;
-				if(dep)
-					this.state.current.addBlock(dep);
+				return true;
 			}
-			return true;
-		}
-	});
-	parser.plugin("call require:amd:array", function(expr, param) {
-		if(param.isArray()) {
-			param.items.forEach(function(param) {
-				var result = this.applyPluginsBailResult("call require:amd:item", expr, param);
-				if(result === undefined) {
-					this.applyPluginsBailResult("call require:amd:context", expr, param);
-				}
-			}, this);
-			return true;
-		} else if(param.isConstArray()) {
-			var deps = [];
-			param.array.forEach(function(request) {
-				var dep, localModule;
-				if(request === "require") {
-					dep = "__webpack_require__";
-				} else if(["exports", "module"].indexOf(request) >= 0) {
-					dep = request;
-				} else if(localModule = LocalModulesHelpers.getLocalModule(this.state, request)) { // eslint-disable-line no-cond-assign
-					dep = new LocalModuleDependency(localModule);
-					dep.loc = expr.loc;
-					this.state.current.addDependency(dep);
+		});
+		parser.plugin("call require:amd:array", (expr, param) => {
+			if(param.isArray()) {
+				param.items.forEach((param) => {
+					const result = parser.applyPluginsBailResult("call require:amd:item", expr, param);
+					if(result === undefined) {
+						parser.applyPluginsBailResult("call require:amd:context", expr, param);
+					}
+				});
+				return true;
+			} else if(param.isConstArray()) {
+				const deps = [];
+				param.array.forEach((request) => {
+					let dep, localModule;
+					if(request === "require") {
+						dep = "__webpack_require__";
+					} else if(["exports", "module"].indexOf(request) >= 0) {
+						dep = request;
+					} else if(localModule = LocalModulesHelpers.getLocalModule(parser.state, request)) { // eslint-disable-line no-cond-assign
```

Comment:
Using another set of parens (to express intent) lets you avoid the `// eslint-disable-line` comment

```js
if ((localModule = LocalModulesHelpers.getLocalModule(parser.state, request))) {
  /* ... */
}
```